napari/layers/shapes/_shapes_mouse_bindings.py

```diff
@@ -437,13 +437,18 @@ def add_path_polygon(layer: Shapes, event: MouseEvent) -> None:
     # on press
     coordinates = layer.world_to_data(event.position)
     if layer._is_creating is False:
+        # Set last cursor position to initial position of the mouse when starting to draw the shape
+        layer._last_cursor_position = np.array(event.pos)
         # Start drawing a path
         initiate_polygon_draw(layer, coordinates)
     else:
         # Add to an existing path or polygon
         index = layer._moving_value[0]
         new_type = Polygon if layer._mode == Mode.ADD_POLYGON else None
-        add_vertex_to_path(layer, event, index, coordinates, new_type)
+        position_diff = np.linalg.norm(event.pos - layer._last_cursor_position)
+        if position_diff:
+            # Ensure the new position is actually a new position
```

Comment:
any reason to use norm instead of just `event.pos == layer._last_cursor_position`? Latter seems more readable to me.

Also, using `position` would be more "correct" than `pos` since it's the world position that shoudln't change, not the canvas pos... In principle we should allow the same canvas pos if it results in a different world pos (i.e: zoom/pan between clicks).

Reply:
Used norm since is used for the polygon lasso tool at https://github.com/napari/napari/blob/26976b57af6a6dff5efc1c3c91bbd5fcee8a542c/napari/layers/shapes/_shapes_mouse_bindings.py#L411-L421

I was not sure if a straight forward comparison could be enough but if that is the case then makes sense to change it indeed :+1: Let me know if I should change it!

The logic uses the `pos` instead of `position` since, from what I checked, seems like that's the value that is being stored as `layer._last_cursor_position`. For example at https://github.com/napari/napari/blob/26976b57af6a6dff5efc1c3c91bbd5fcee8a542c/napari/layers/shapes/_shapes_mouse_bindings.py#L339-L340 and https://github.com/napari/napari/blob/26976b57af6a6dff5efc1c3c91bbd5fcee8a542c/napari/layers/shapes/_shapes_mouse_bindings.py#L388

Should I try to change in this PR the usage of `pos` for `position` for the `layer._last_cursor_position` logic?

Reply:
The norm makes sense in that place because there's a threshold; in this case, you're checking only if the vertex is *identical*, so I would use `==` (probably faster as well, though I doubt it matters :P).

Ah, I see; I thought the `layer._last_cursor_position` was a new thing you created here. I wouldn't change the rest then.

However, this makes me wonder if there's any reason for this value to be stored on the layer and shared between different mouse callbacks? Or worse, can it be a problem if they fight over this value? Maybe there's no need to address in this PR, but we definitely need to make sure we don't end up messing up something else by touching this value.

Reply:
> However, this makes me wonder if there's any reason for this value to be stored on the layer and shared between different mouse callbacks? Or worse, can it be a problem if they fight over this value? Maybe there's no need to address in this PR, but we definitely need to make sure we don't end up messing up something else by touching this value.

Thinking about this, I would said that the value is stored over the layer so it can be retrieved later to check the difference (in the case of the polygon lasso tool) and now the equivalence (for the polygon tool). Checking, I think that the value `layer._last_cursor_position` is not being set to `None` when finishing the drawing, so maybe it could be worthy to add such a "reset" (`layer._last_cursor_position = None`) as a way to mitigate possible interference between usages when drawing a shape finishes?

Reply:
yeah, good idea! Ideally this would be tracke dby the mouse callback itself... but I guess it's not that easy ^^'